### ISSUE TITLE
Infinite storage battery bug by installing a 'swappable storage battery' with a small or medium storage battery

### ISSUE THREAD

**Issue by Cosmitz** [CleverRaven/Cataclysm-DDA#21698]
https://streamable.com/ew1kc

Having a small storage battery or a medium storage battery in a valid location (inventory/ground) fulfills the requirements for installing a 'swappable storage battery' in a storage battery case location on the vehicle. That fills said vehicle location with a "swappable storage battery" and doesn't consume the small or medium batteries. At this point i can remove the installed "swappable storage battery" and it will be extracted to the world as a "storage battery" (seemingly also randomly full or empty).

I can do this to get infinite storage batteries.

A fix for this should support installing different types of batteries and extracting the correct type.

Ran into this when i wanted to actually make small and medium storage batteries work as separate vehicle parts. Realised they're already in the game.. sort of.. kind of. ish.

**Comment by kevingranade** [comment]
Please describe the issue, "watch this video and guess what the problem is" isn't a valid bug report.

**Comment by Cosmitz** [comment]
Having a small storage battery or a medium storage battery in a valid location fulfills the requirements for installing a 'swappable storage battery' in a storage battery case location. That fills said vehicle location with a "swappable storage battery" and doesn't consume the small or medium batteries. At this point i can remove the installed "swappable storage battery" and it will be extracted to the world as a "storage battery", seemingly also randomly full or empty. 

I can do this to get infinite storage batteries.

**Comment by sxmichaels** [comment]
I just tried what @Cosmitz  did in the video, with small and medium storage batteries on both the ground and in my inventory, and looked at what I could install in a tile with a `swappable storage battery case`. Neither one qualified as a `storage battery` and I was not able to install. I think this bug has been fixed.

<img width="476" alt="2017-12-05 20 44 16" src="https://user-images.githubusercontent.com/13933679/33645010-6199efe0-d9fd-11e7-811c-3648baadd1de.png">
<img width="823" alt="2017-12-05 20 43 47" src="https://user-images.githubusercontent.com/13933679/33645009-618439e8-d9fd-11e7-9a22-ee98803f52a5.png">
